src/bin/cql-stress-cassandra-stress/settings/command/mixed.rs

```diff
@@ -0,0 +1,94 @@
+use std::collections::HashSet;
+
+use crate::{
+    java_generate::distribution::enumerated::EnumeratedDistribution,
+    settings::param::types::Parsable,
+};
+use anyhow::{Context, Result};
+
+use super::Command;
+
+// Available subcommands for mixed command.
+#[derive(Copy, Clone, Hash, PartialEq, Eq)]
+pub enum MixedSubcommand {
+    Read,
+    Write,
+    CounterRead,
+    CounterWrite,
+}
+
+impl std::fmt::Display for MixedSubcommand {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match self {
+            MixedSubcommand::Read => "read",
+            MixedSubcommand::Write => "write",
+            MixedSubcommand::CounterRead => "counter_read",
+            MixedSubcommand::CounterWrite => "counter_write",
+        };
+        write!(f, "{}", s)
+    }
+}
+
+pub type OperationRatio = EnumeratedDistribution<MixedSubcommand>;
+
+impl OperationRatio {
+    fn parse_command_weight(s: &str) -> Result<(MixedSubcommand, f64)> {
+        let (cmd, weight) = {
+            let mut iter = s.split('=').fuse();
+            match (iter.next(), iter.next(), iter.next()) {
+                (Some(cmd), Some(w), None) => (cmd, w),
+                _ => anyhow::bail!(
+                    "Command weight specification should match pattern <command>=<f64>"
+                ),
+            }
+        };
+
+        let command = match Command::parse(cmd)? {
+            Command::Read => MixedSubcommand::Read,
+            Command::Write => MixedSubcommand::Write,
+            Command::CounterRead => MixedSubcommand::CounterRead,
+            Command::CounterWrite => MixedSubcommand::CounterWrite,
+            _ => anyhow::bail!("Invalid command for mixed workload: {}", cmd),
+        };
+        let weight = weight.parse::<f64>()?;
+        Ok((command, weight))
+    }
+
+    fn do_parse(s: &str) -> Result<Self> {
+        // Remove wrapping parenthesis.
+        let arg = {
+            let mut chars = s.chars();
+            anyhow::ensure!(
+                chars.next() == Some('(') && chars.next_back() == Some(')'),
+                "Invalid operation ratio specification: {}",
+                s
+            );
+            chars.as_str()
+        };
+
+        let mut command_set = HashSet::<MixedSubcommand>::new();
+        let weights = arg
+            .split(',')
+            .map(|s| -> Result<(MixedSubcommand, f64)> {
+                let (command, weight) = Self::parse_command_weight(s)?;
+                anyhow::ensure!(
+                    !command_set.contains(&command),
+                    "{} command has been specified more than once",
+                    command
+                );
+                command_set.insert(command);
+                Ok((command, weight))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        Self::new(&weights)
+    }
+}
```

Comment:
Side note: this falls under the "custom syntax" mentioned in https://github.com/scylladb/cql-stress/issues/52 (no need to do anything right now).